### PR TITLE
Add datagen tools

### PR DESCRIPTION
**Pull request by lumurillo**
The goal of this PR is to include a set of tools useful for the creation of COCO- or MOTS's compatible datasets, capturing the data from the CARLA simulator.

### CODE REVIEW

File: lib/oscar-datagen-toolkit/Makefile
[the remark this answers is not in the document]
Comment (dxoigmn) [reply lib/oscar-datagen-toolkit/Makefile:1]: I think replacing this with docker compose is probably the right call.

**Review by dxoigmn — COMMENTED** [review]
See comments. It might also be worth reviewing this paper and code: https://arxiv.org/abs/2305.00204

File: lib/oscar-datagen-toolkit/Makefile
[the remark this answers is not in the document]
Comment (lumurillo) [reply lib/oscar-datagen-toolkit/Makefile:1]: Working on adding docker compose support

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  # word spelling linter
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        args:
+          - --skip=logs/**,data/**
+          - --ignore-words-list=abc,def,gard,ue
+
```
Comment (dxoigmn) [inline .pre-commit-config.yaml:103]: Can you fix the newline issue

File: lib/oscar-datagen-toolkit/.env [hunk 2]
```diff
@@ -0,0 +1,7 @@
+DATAGEN_TOOLS_VERSION=3.0.0
+CARLA_VERSION=0.9.13
+CARLA_PORT=5000
+CARLA_TM_PORT=5002
+HOST_MOUNT_POINT=./data
+U_ID=0
+G_ID=0
```
Comment (dxoigmn) [inline lib/oscar-datagen-toolkit/.env:7]: UID and GID are probably more standard linux names.

**Review by dxoigmn — CHANGES_REQUESTED** [review]
See comments. Can you also change the headers to be BSD-3 license SPDX identifier?

File: lib/oscar-datagen-toolkit/.env (hunk 2 above)
Comment (lumurillo) [reply lib/oscar-datagen-toolkit/.env:7]: Linux already has an env variable called `UID`, so I wanted to make this variables independent to that.

File: lib/oscar-datagen-toolkit/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+> conda activate myenv
+>
+> # [OPTIONAL] create virtualenv environment
+> python3.9 -m venv .venv
+> source .venv/bin/activate
+> ```
+
+## Getting started
+
+- With Docker:
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply lib/oscar-datagen-toolkit/README.md]: yes, agree

File: lib/oscar-datagen-toolkit/README.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+# Build oscar/datagen image
+docker compose build annotator
+
+# Build CARLA image
+docker compose build carla
+```
+
+> **Note**: The `collect.sh` and `annotate.sh` scripts automatically verify the exitances of these Docker images and build them if necessary. You can use the above command if you want to rebuild the Docker images.
+
+- Manual installation
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply lib/oscar-datagen-toolkit/README.md]: This was maybe a bad translation from spanish. I changed it to `Install from source code`

File: lib/oscar-datagen-toolkit/README.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
+# OSCAR Datagen Toolkit
+
+This repository contains a set of tools for generating synthetic datasets using the CARLA Simulator. [CARLA](https://carla.org/) is an open-source simulator for autonomous driving research that provides realistic urban environments, dynamic traffic, and realistic sensors.
+
```
[the remark this answers is not in the document]
Comment (lumurillo) [reply lib/oscar-datagen-toolkit/README.md:4]: Added

**Review by dxoigmn — APPROVED** [review]
LGTM!

**Pull request merged**